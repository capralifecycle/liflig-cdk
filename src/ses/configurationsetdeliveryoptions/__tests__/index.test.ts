import { App, Stack } from "aws-cdk-lib"
import "jest-cdk-snapshot"
import { ConfigurationSetDeliveryOptions } from "../index"

test("configuration-set-delivery-options", () => {
  const app = new App()
  const stack = new Stack(app, "Stack")

  new ConfigurationSetDeliveryOptions(stack, "DeliveryOptions", {
    configurationSetName: "exampleconfigurationset",
    tlsPolicy: "Require",
  })

  expect(stack).toMatchCdkSnapshot({
    ignoreAssets: true,
  })
})
