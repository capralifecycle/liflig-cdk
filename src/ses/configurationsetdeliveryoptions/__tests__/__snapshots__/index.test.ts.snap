// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`configuration-set-delivery-options 1`] = `
Object {
  "Mappings": Object {
    "DefaultCrNodeVersionMap": Object {
      "af-south-1": Object {
        "value": "nodejs16.x",
      },
      "ap-east-1": Object {
        "value": "nodejs16.x",
      },
      "ap-northeast-1": Object {
        "value": "nodejs16.x",
      },
      "ap-northeast-2": Object {
        "value": "nodejs16.x",
      },
      "ap-northeast-3": Object {
        "value": "nodejs16.x",
      },
      "ap-south-1": Object {
        "value": "nodejs16.x",
      },
      "ap-south-2": Object {
        "value": "nodejs16.x",
      },
      "ap-southeast-1": Object {
        "value": "nodejs16.x",
      },
      "ap-southeast-2": Object {
        "value": "nodejs16.x",
      },
      "ap-southeast-3": Object {
        "value": "nodejs16.x",
      },
      "ca-central-1": Object {
        "value": "nodejs16.x",
      },
      "cn-north-1": Object {
        "value": "nodejs16.x",
      },
      "cn-northwest-1": Object {
        "value": "nodejs16.x",
      },
      "eu-central-1": Object {
        "value": "nodejs16.x",
      },
      "eu-central-2": Object {
        "value": "nodejs16.x",
      },
      "eu-north-1": Object {
        "value": "nodejs16.x",
      },
      "eu-south-1": Object {
        "value": "nodejs16.x",
      },
      "eu-south-2": Object {
        "value": "nodejs16.x",
      },
      "eu-west-1": Object {
        "value": "nodejs16.x",
      },
      "eu-west-2": Object {
        "value": "nodejs16.x",
      },
      "eu-west-3": Object {
        "value": "nodejs16.x",
      },
      "me-central-1": Object {
        "value": "nodejs16.x",
      },
      "me-south-1": Object {
        "value": "nodejs16.x",
      },
      "sa-east-1": Object {
        "value": "nodejs16.x",
      },
      "us-east-1": Object {
        "value": "nodejs16.x",
      },
      "us-east-2": Object {
        "value": "nodejs16.x",
      },
      "us-gov-east-1": Object {
        "value": "nodejs16.x",
      },
      "us-gov-west-1": Object {
        "value": "nodejs16.x",
      },
      "us-iso-east-1": Object {
        "value": "nodejs14.x",
      },
      "us-iso-west-1": Object {
        "value": "nodejs14.x",
      },
      "us-isob-east-1": Object {
        "value": "nodejs14.x",
      },
      "us-west-1": Object {
        "value": "nodejs16.x",
      },
      "us-west-2": Object {
        "value": "nodejs16.x",
      },
    },
  },
  "Resources": Object {
    "AWS679f53fac002430cb0da5b7982bd22872D164C4C": Object {
      "DependsOn": Array [
        "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
      ],
      "Properties": Object {
        "Code": Any<Object>,
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
            "Arn",
          ],
        },
        "Runtime": Object {
          "Fn::FindInMap": Array [
            "DefaultCrNodeVersionMap",
            Object {
              "Ref": "AWS::Region",
            },
            "value",
          ],
        },
        "Timeout": 120,
      },
      "Type": "AWS::Lambda::Function",
    },
    "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "DeliveryOptionsCustomResourcePolicyBEF2DFD1": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "ses:PutConfigurationSetDeliveryOptions",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "DeliveryOptionsCustomResourcePolicyBEF2DFD1",
        "Roles": Array [
          Object {
            "Ref": "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "DeliveryOptionsDC6E2A1D": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "DeliveryOptionsCustomResourcePolicyBEF2DFD1",
      ],
      "Properties": Object {
        "Create": "{\\"service\\":\\"SES\\",\\"action\\":\\"putConfigurationSetDeliveryOptions\\",\\"parameters\\":{\\"ConfigurationSetName\\":\\"exampleconfigurationset\\",\\"DeliveryOptions\\":{\\"TlsPolicy\\":\\"Require\\"}},\\"physicalResourceId\\":{\\"id\\":\\"exampleconfigurationset\\"}}",
        "Delete": "{\\"service\\":\\"SES\\",\\"action\\":\\"putConfigurationSetDeliveryOptions\\",\\"parameters\\":{\\"ConfigurationSetName\\":\\"exampleconfigurationset\\"}}",
        "InstallLatestAwsSdk": true,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "AWS679f53fac002430cb0da5b7982bd22872D164C4C",
            "Arn",
          ],
        },
        "Update": "{\\"service\\":\\"SES\\",\\"action\\":\\"putConfigurationSetDeliveryOptions\\",\\"parameters\\":{\\"ConfigurationSetName\\":\\"exampleconfigurationset\\",\\"DeliveryOptions\\":{\\"TlsPolicy\\":\\"Require\\"}},\\"physicalResourceId\\":{\\"id\\":\\"exampleconfigurationset\\"}}",
      },
      "Type": "Custom::AWS",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
